ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -44,7 +44,8 @@
   public final Counter selectorKeyOperationErrorCount;
   public final Counter selectorCloseKeyErrorCount;
   public final Counter selectorCloseSocketErrorCount;
-  public Gauge<Long> selectorActiveConnections;
+  public Gauge<Long> numActiveConnections;
+  public Gauge<Long> numConnectionsPendingHandshake;
```

Comment:
This should not be a `selector` metric as this is not "general". This should be moved below under SSL Metrics and `SSLTransmission` should be the one updating it.

That said, I would suggest getting rid of this metric completely. This is a `Gauge` and number of connections pending handshake at a particular instant is not useful as these will be very low in number and they would either error out or get established. `Gauge` would be sort of useful for things that stay for a while like `numActiveConnections`.